modules/statistics/statistics.js

```diff
@@ -230,34 +247,89 @@ Statistics.prototype.stopRemoteStats = function() {
 
 /**
  * Initializes the callstats.io API.
- * @param peerConnection {JingleSessionPC} the session object
+ * @param {TraceablePeerConnection} tpc the {@link TraceablePeerConnection}
+ * instance for which CalStats will be started.
+ * @param {string} remoteUserID
  */
-Statistics.prototype.startCallStats = function(session) {
-    if (this.callStatsIntegrationEnabled && !this.callStatsStarted) {
-        // Here we overwrite the previous instance, but it must be bound to
-        // the new PeerConnection
-        this.callstats = new CallStats(session, this.options);
-        Statistics.callsStatsInstances.push(this.callstats);
-        this.callStatsStarted = true;
+Statistics.prototype.startCallStats = function(tpc, remoteUserID) {
+    if (!this.callStatsIntegrationEnabled) {
+        return;
+    } else if (this.callsStatsInstances.has(tpc.id)) {
+        logger.error('CallStats instance for ${tpc} exists already');
+
+        return;
+    }
+
+    if (!CallStats.isBackendInitialized()) {
```

Comment:
Not sure if I understand what you mean, but throwing from a constructor only because it's called with different arguments seems a bit weird to me.

I think that it may be possible to initialise backend another time, but then the previous one will be closed (there will be new WebSocket connection to the Callstats server ?). But I'm not sure about that. If that would be the case still I would prefer to add method 'disposeBackend' and have that called from outside, so that this action would be explicit, because it would overwrite the previous backend. It should be "informed decision" and only on the outside of CallStats we have enough information to take such decision.